package.json

```diff
@@ -64,6 +68,8 @@
     "klona": "^2.0.6",
     "mocha": "^10.7.0",
     "mocha-lcov-reporter": "^1.3.0",
+    "rimraf": "^6.0.1",
```

Comment:
is rimraf really a main dependency? https://www.npmjs.com/package/rimraf

This seems like an entry in `devDepencies` also do `@types/node` and `typescript` have to be in the dependencies for all users? It seems like that introduces a lot of bloat in this library. Are you sure those aren't `devDependencies` also?

Reply:
Sorry, those definitively belong in the `devDependencies`. You are also right that `rimraf` is not a necessary dependency and only used during development to clear up between builds. You could also use the usual `rm` command on linux, but then it would not be cross-plattform compatible.

Reply:
- The `rimraf` dependency is only for the convenience of the developer, so that previous builds must not be cleared manually during development. You can see it's usage in the package.json-`build-types`-script.
- The `typescript` dependency is obviously necessary to build the types. You can see usage of `tsc` in the `build-types`-script.
- The `@types/node` dependency is used to give typescript the necessary types in the NodeJS environment, so that it will not warn of missing types and validate the usage of internal classes and API.

These are just the basics of typescript development and should not really be a surprise. If you install the package in production those dependencies will not even be downloaded. The published types should then already be included in the package. Alternatively you can publish proper types in the [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repository, but then you should not just use automatically generated types.